### PR TITLE
remove websocket_error event, fixes #4674

### PR DESCRIPTION
**Pull request by Prinzhorn**
#### Description

I think we need a filter for close_code now.

Original discussion in #4674

#### Checklist

 - [x] I have updated tests where applicable.

### CODE REVIEW

**Comment by mhils** [comment]
Your argument in https://github.com/mitmproxy/mitmproxy/pull/4674#issuecomment-878297628 sounds good to me. This looks good to merge except for the missing coverage in `dumper.py`. Can you handle that soon-ish? Otherwise I'll take care of this so that we can release 7.0 this week :)

File: test/mitmproxy/proxy/layers/test_websocket.py [hunk 1]
```diff
@@ -245,11 +245,10 @@ def test_close_disconnect(ws_testdata):
             << CloseConnection(tctx.server)
             << SendData(tctx.client, b"\x88\x02\x03\xe8")
             << CloseConnection(tctx.client)
-            << websocket.WebsocketErrorHook(flow)
+            << websocket.WebsocketEndHook(flow)
             >> reply()
             >> ConnectionClosed(tctx.client)
     )
-    assert "ABNORMAL_CLOSURE" in flow.error.msg
```
Comment (mhils) [inline test/mitmproxy/proxy/layers/test_websocket.py:252]: Can we please assert the code here?

File: test/mitmproxy/proxy/layers/test_websocket.py [hunk 2]
```diff
@@ -263,10 +262,9 @@ def test_close_error(ws_testdata):
             << CloseConnection(tctx.server)
             << SendData(tctx.client, b"\x88\x02\x0f\xa0")
             << CloseConnection(tctx.client)
-            << websocket.WebsocketErrorHook(flow)
+            << websocket.WebsocketEndHook(flow)
             >> reply()
     )
-    assert "UNKNOWN_ERROR=4000" in flow.error.msg
```
Comment (mhils) [inline test/mitmproxy/proxy/layers/test_websocket.py:269]: and here?

**Review by mhils — APPROVED** [review]
Let's add the assertions and then please feel free to merge yourself. 😃

**Comment by Prinzhorn** [comment]
@mhils please check my recent commit. Did I understand that correctly (I'm still having a hard time understanding playbooks)? I think `test_close_error` is more like `test_application_close_code` now, because we're no longer populating `flow.error` and hence this basically tests if wsproto supports arbitrary codes in the 4xxx range?

**Comment by mhils** [comment]
Yes this looks good to me. `test_close_error` is making sure that we propagate the error code to the other side if one side sends a close. In contrast, `test_close_disconnect` makes sure that we send a disconnect if the peer completely disconnects without sending anything.

Feel free to change the code to something that's definitely an error or rename the function, either is fine with me :)

**Comment by Prinzhorn** [comment]
Welp, first time merging a PR and I was expecting it to squash automatically :shrug:

**Comment by mhils** [comment]
All good - you have to break an egg to make an omelet. :)

**Pull request merged**